invenio_rdm_records/assets/semantic-ui/js/invenio_rdm_records/oaipmh/details/LinksTable.js

```diff
@@ -5,53 +5,166 @@
 // under the terms of the MIT License; see LICENSE file for more details.
 
 import PropTypes from "prop-types";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { i18next } from "@translations/invenio_administration/i18next";
 import CopyButton from "./CopyButton";
-import { Table, Header } from "semantic-ui-react";
+import { Table, Header, Dropdown, Grid } from "semantic-ui-react";
+import { http } from "@js/invenio_administration/src/api/config.js";
+import { capitalize } from "lodash";
+
+/** Map of known formats and their name. */
+const knownFormats = {
+  "oai_dc": "OAI Dublin Core",
+  "datacite": "DataCite",
+  "oai_datacite": "OAI DataCite",
+}
 
 function LinksTable({ data }) {
+  const [links, setLinks] = useState(data.links);
```

Comment:
This func is massive! :) I would suggest splitting it. Does it make sense to keep a functional cmp here?